File: 프로그래머스/lv2/12941. 최솟값 만들기/최솟값 만들기.py
def solution(A,B):
    answer = 0
    
    # for _ in range(len(A)):
        # answer += A.pop(A.index(min(A))) * B.pop(B.index(max(B)))       # 효율성 테스트 실패
        
    A.sort()
    B.sort(reverse = True)
    for a, b in zip(A, B):
        answer += a * b

    return answer

File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.java
import java.util.Arrays;

class Solution
{
    public int solution(int []A, int []B)
    {
        int answer = 0;
        int len = A.length;
        Arrays.sort(A);
        Arrays.sort(B);
        
        for (int i = 0; i < len; i++) {
            answer += (A[i] * B[len - i - 1]);        // 제일 큰 값 * 제일 작은 값
        }

        return answer;
    }
}